### PR TITLE
feat(paths): XDG paths + branch sanitizer + path template renderer

### LINKED ISSUE #3: vs-01b: XDG paths + branch sanitizer + path template renderer
## Parent PRD

#1

## What to build

Implement three foundational utilities in `src/paths.rs`:

1. **XDG directory resolution** — functions returning config (`~/.config/trench/`), data (`~/.local/share/trench/`), state (`~/.local/state/trench/`), and worktree root (`~/.worktrees/`) paths using the `dirs` crate. Each function creates the directory if it doesn't exist.

2. **Branch name sanitizer** — converts branch names to filesystem-safe directory names: `/` → `-`, spaces → `-`, `@` → `-`, `..` → stripped, consecutive dashes collapsed, single dots preserved (e.g., `v2.1.3` stays). See FR-15, FR-16.

3. **Path template renderer** — uses `minijinja` to render worktree path templates with `{{ repo }}` and `{{ branch | sanitize }}` tokens. Default template: `~/.worktrees/{{ repo }}/{{ branch | sanitize }}`. See FR-17.

### Layers touched
paths, config, tests

## Acceptance criteria

- [ ] `config_dir()`, `data_dir()`, `state_dir()`, `worktree_root()` return correct XDG paths
- [ ] All path functions use `dirs` crate (not hardcoded)
- [ ] Directories are created if they don't exist (ensure_dir pattern)
- [ ] `sanitize_branch("feature/auth")` → `"feature-auth"`
- [ ] `sanitize_branch("fix@home")` → `"fix-home"`
- [ ] `sanitize_branch("a..b")` → `"a-b"` (double dots stripped)
- [ ] `sanitize_branch("a--b")` → `"a-b"` (consecutive dashes collapsed)
- [ ] `sanitize_branch("v2.1.3")` → `"v2.1.3"` (single dots preserved)
- [ ] Template renderer resolves `{{ repo }}` and `{{ branch | sanitize }}`
- [ ] Unit tests cover all sanitization edge cases and path resolution

## Blocked by

- #2

## User stories addressed

- US-1 (worktree path resolution)
- US-7 (opinionated path convention)

### PR DESCRIPTION
**Pull request by sadiksaifi**
Closes #3

## Summary
Implements three foundational utilities in `src/paths.rs`: XDG-compliant directory resolution functions (`config_dir`, `data_dir`, `state_dir`, `worktree_root`) using the `dirs` crate with automatic directory creation, a branch name sanitizer that converts branch names to filesystem-safe directory names per FR-15/FR-16, and a path template renderer using minijinja with `{{ repo }}` and `{{ branch | sanitize }}` support per FR-17.

## User Stories Addressed
- US-1: Create a new worktree for a feature branch with one command (worktree path resolution)
- US-7: Batch-sync all worktrees with a single command (opinionated path convention)

## Automated Testing
- Total tests added: 15
- Tests passing: 15
- Tests failing: 0
- `cargo clippy`: pass
- `cargo test`: pass (23 total tests, 15 new + 8 existing)

## Manual Testing Checklist
1. Clone the repo and checkout this branch
2. Run `cargo test paths::tests` — all 15 tests should pass
3. Run `cargo clippy --tests` — no warnings beyond expected dead_code (module not yet wired into main)
4. Verify XDG directories were created: `ls ~/.config/trench/ ~/.local/share/trench/ ~/.local/state/trench/ ~/.worktrees/`
5. Open `src/paths.rs` and verify `sanitize_branch` handles: slashes, @, spaces, double dots, consecutive dashes, single dots preserved
6. Verify `render_worktree_path` with default template produces `repo/sanitized-branch` format
7. Run `cargo test` — all 23 tests (8 existing CLI + 15 new paths) should pass

## Known Issues
None

## Summary by CodeRabbit

* **New Features**
  * Added standardized directory management for config, data, and state with automatic creation and environment-aware resolution.
  * Added worktree path templating with a default template and branch name sanitization to produce safe filesystem paths.

* **Tests**
  * Added unit tests validating directory resolution, template rendering, and branch sanitizer behavior.

### CODE REVIEW

File: src/paths.rs [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+/// Return the trench state directory (`~/.local/state/trench/`), creating it if needed.
+pub fn state_dir() -> Result<PathBuf> {
+    let base = dirs::home_dir()
+        .context("could not determine home directory")?
+        .join(".local")
+        .join("state");
+    let path = base.join(APP_NAME);
+    ensure_dir(&path)?;
+    Ok(path)
+}
```
[the remark this answers is not in the document]
Comment (sadiksaifi) [reply src/paths.rs:52]: @coderabbitai Addressed in b810cdf — `state_dir()` now tries `dirs::state_dir()` first and falls back to `~/.local/state` via a `STATE_DIR_FALLBACK_SEGMENTS` constant for macOS/Windows where `dirs::state_dir()` returns `None`.

File: src/paths.rs [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+/// Rules (FR-15, FR-16):
+/// - `/` → `-`
+/// - spaces → `-`
+/// - `@` → `-`
+/// - `..` → stripped
+/// - consecutive dashes collapsed
+/// - single dots preserved
+pub fn sanitize_branch(branch: &str) -> String {
+    // Replace `..` sequences (path traversal) with dash
+    let stripped = branch.replace("..", "-");
```
[the remark this answers is not in the document]
Comment (sadiksaifi) [reply src/paths.rs:103]: @coderabbitai Fixed in b2ed893 — doc comment now reads `.. → -` to match the actual replacement behavior.

**Comment by sadiksaifi** [comment]
@coderabbitai Edge case tests for `sanitize_branch` (empty string, single dot, triple dots) added in 280ab37.

File: src/paths.rs [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+}
+
+/// Return the trench state directory (`~/.local/state/trench/`), creating it if needed.
+///
+/// Uses `dirs::state_dir()` when available (Linux), falls back to
+/// `~/.local/state` on platforms that return `None` (macOS/Windows).
+pub fn state_dir() -> Result<PathBuf> {
+    let base = dirs::state_dir().unwrap_or_else(|| {
+        let home = dirs::home_dir().expect("could not determine home directory");
+        STATE_DIR_FALLBACK_SEGMENTS.iter().fold(home, |p, s| p.join(s))
```
[the remark this answers is not in the document]
Comment (sadiksaifi) [reply src/paths.rs]: @coderabbitai Fixed in 2895d45 — replaced `expect()` with `match` + `context("...")?.` so a missing home directory returns `Err` instead of panicking.

File: src/paths.rs [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    let mut env = minijinja::Environment::new();
+    env.add_filter("sanitize", sanitize_branch);
+    env.add_template("path", template)
+        .context("invalid worktree path template")?;
+    let tmpl = env.get_template("path").unwrap();
+    let rendered = tmpl
+        .render(minijinja::context! { repo => repo, branch => branch })
+        .context("failed to render worktree path template")?;
+    Ok(PathBuf::from(rendered))
+}
```
[the remark this answers is not in the document]
Comment (sadiksaifi) [reply src/paths.rs:90]: @coderabbitai Fixed in b8ba1dd — `render_worktree_path` now rejects absolute paths and `..` segments with `anyhow::bail\!`. Added two tests: `render_template_rejects_absolute_path` and `render_template_rejects_parent_dir`.

**Comment by sadiksaifi** [comment]
@coderabbitai reivew - is it ready to merge
